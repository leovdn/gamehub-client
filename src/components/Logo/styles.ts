import styled, { css } from 'styled-components'
import { LogoProps } from '.'

const wrapperModifiers = {
  normal: () => css`
    width: 11rem;
    height: 2rem;
  `,
  large: () => css`
    width: 20rem;
    height: 4rem;
  `,

  hideonmobile: () => css`
    @media (max-width: 600px) {
      width: 5.8rem;
      height: 2.5rem;

      svg {
        height: 3.2rem;
        pointer-events: none;
      }

      .text {
        display: none;
      }
    }
  `
}

export const Wrapper = styled.div<LogoProps>`
  ${({ theme, color, size, hideonmobile }) => `
    color: ${theme.colors[color!]};

    ${!!size && wrapperModifiers[size]()}
    ${!!hideonmobile && wrapperModifiers.hideonmobile()}
  `}
`
