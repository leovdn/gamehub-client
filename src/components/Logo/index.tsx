import Image from 'next/image'
import LogoImg from '../../../public/img/logo-outline.svg'
import * as S from './styles'

const Logo = () => {
  return (
    <S.Wrapper>
      <Image
        src={LogoImg}
        alt="Logo with a joystick writen Gamehub on the right"
      />
    </S.Wrapper>
  )
}

export default Logo
