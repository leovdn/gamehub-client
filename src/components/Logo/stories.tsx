import Logo from '.'
import { Meta, StoryFn } from '@storybook/react'

export default {
  title: 'Logo',
  component: Logo,
  args: {
    title: '',
    description: ''
  }
} as Meta

export const Basic: StoryFn = (args) => <Logo {...args} />
