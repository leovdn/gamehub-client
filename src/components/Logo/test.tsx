import { render } from '@testing-library/react'

import Logo from './index'

describe('<Logo />', () => {
  it('should render the heading', () => {
    const { container } = render(<Logo />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
