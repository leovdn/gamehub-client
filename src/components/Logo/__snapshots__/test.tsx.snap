// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the heading 1`] = `
<main
  class=""
>
  <img
    alt="Logo with a joystick writen Gamehub on the right"
    data-nimg="1"
    decoding="async"
    src=""
    style="color: transparent;"
  />
</main>
`;
